heron/executor/src/python/heron_executor.py

```diff
@@ -325,6 +337,47 @@ def _get_metricsmgr_cmd(self, metricsManagerId, sink_config_file, port):
 
     return metricsmgr_cmd
 
+  def _get_metrics_cache_cmd(self):
+    ''' get the command to start the metrics manager processes '''
+    metricscachemgr_main_class = 'com.twitter.heron.metricscachemgr.MetricsCacheManager'
+
+    metricscachemgr_cmd = [os.path.join(self.heron_java_home, 'bin/java'),
+                           # We could not rely on the default -Xmx setting, which could be very big,
+                           # for instance, the default -Xmx in Twitter mesos machine is around 18GB
+                           '-Xmx1024M',
```

Comment:
The container 0 RAM restriction is as same as the other containers RAM, which can be determined by self.component_rammap. However, the tmaster process does not have ram limit configured. The RAM limit for the other processes in the container 0 cannot be determined. The 1024M is the same value for the java process MetricsMgr in container 0.

tmaster(RAM unknown) + MetricsMgr(1024M) + MetricsCacheMgr(1024M) = container 0 RAM